indexer/extension/evmret/extension.go

```diff
@@ -0,0 +1,161 @@
+	if cfg.GetVmType() != types.EVM || !cfg.EvmRetCleanupEnabled() {
+		return nil
+	}
+
+	return &EvmRetCleanupExtension{
+		cfg:    cfg,
+		logger: logger.With("extension", ExtensionName),
+		db:     db,
+	}
+}
+
+// Name returns the name of the extension
+func (e *EvmRetCleanupExtension) Name() string {
+	return ExtensionName
+}
+
+// Initialize ensures the status table exists and retrieves current status
+func (e *EvmRetCleanupExtension) Initialize(ctx context.Context) (*types.CollectedEvmRetCleanupStatus, error) {
+	// Create status table if it doesn't exist
+	if err := e.db.WithContext(ctx).AutoMigrate(&types.CollectedEvmRetCleanupStatus{}); err != nil {
+		return nil, fmt.Errorf("failed to create status table: %w", err)
+	}
+
+	// Try to retrieve existing status
+	var status types.CollectedEvmRetCleanupStatus
+	err := e.db.WithContext(ctx).First(&status).Error
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			// Create initial status record
+			status = types.CollectedEvmRetCleanupStatus{
+				LastCleanedHeight: 0,
+				CorrectedRecords:  0,
+			}
+			if err := e.db.WithContext(ctx).Create(&status).Error; err != nil {
+				return nil, fmt.Errorf("failed to create initial status: %w", err)
+			}
+			e.logger.Info("Initialized cleanup status")
+			return &status, nil
+		}
+		return nil, fmt.Errorf("failed to retrieve cleanup status: %w", err)
+	}
+
+	return &status, nil
+}
+
+// Run executes the cleanup extension continuously
+func (e *EvmRetCleanupExtension) Run(ctx context.Context) error {
+	// Initialize and get status
+	status, err := e.Initialize(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to initialize: %w", err)
+	}
+
+	currentHeight := status.LastCleanedHeight + 1
+	e.logger.Info("starting EVM ret cleanup",
+		slog.Int64("start_height", currentHeight))
+
+	// Continuous processing loop
+	for {
+		// Check for context cancellation
+		select {
+		case <-ctx.Done():
+			e.logger.Info("cleanup stopped",
+				slog.Int64("last_cleaned_height", status.LastCleanedHeight),
+				slog.Int64("corrected_records", status.CorrectedRecords))
+			return ctx.Err()
+		default:
+		}
+
+		// Get the maximum height from the tx table
+		var maxHeight int64
+		if err := e.db.WithContext(ctx).
+			Model(&types.CollectedTx{}).
+			Select("COALESCE(MAX(height), 0)").
+			Scan(&maxHeight).Error; err != nil {
+			return fmt.Errorf("failed to get max height: %w", err)
+		}
+
+		// Wait if we've caught up to the current max height
+		if currentHeight > maxHeight {
+			// Small, context-aware backoff to avoid busy-polling the DB
+			select {
+			case <-ctx.Done():
+				e.logger.Info("cleanup stopped",
+					slog.Int64("last_cleaned_height", status.LastCleanedHeight),
+					slog.Int64("corrected_records", status.CorrectedRecords))
+				return ctx.Err()
+			case <-time.After(1 * time.Second):
+			}
+			continue
+		}
+
+		endHeight := currentHeight + BatchSize - 1
```

Comment:
Since newer Go versions have min and max, we could just use those. Much easier to read.

Reply:
lol